satellite/metainfo/metainfo.go

```diff
@@ -511,47 +514,63 @@ func CreatePath(ctx context.Context, projectID uuid.UUID, segmentIndex int64, bu
 func (endpoint *Endpoint) SetAttribution(ctx context.Context, req *pb.SetAttributionRequest) (_ *pb.SetAttributionResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	_, err = endpoint.checkBucketPointers(ctx, req)
-	if err != nil {
-		// TODO: return correct status code for GRPC
-		endpoint.log.Sugar().Debug("related bucket id already attributed \n")
-		return &pb.SetAttributionResponse{}, err
-	}
-
-	// TODO: add valueattribution DB access functions added in new PR.
-
-	return &pb.SetAttributionResponse{}, nil
-}
-
-// checks if bucket has any pointers(entries)
-func (endpoint *Endpoint) checkBucketPointers(ctx context.Context, req *pb.SetAttributionRequest) (resp bool, err error) {
-	//TODO: Logic of checking if bucket exists will be added in new PR.
-	// write into value attribution DB only if bucket exists but no segments or no bucket and no segments exits
-	defer mon.Task()(&ctx)(&err)
-
 	keyInfo, err := endpoint.validateAuth(ctx, macaroon.Action{
 		Op:            macaroon.ActionList,
 		Bucket:        req.BucketName,
 		EncryptedPath: []byte(""),
 		Time:          time.Now(),
 	})
 	if err != nil {
-		return false, status.Errorf(codes.Unauthenticated, err.Error())
+		return nil, status.Errorf(codes.Unauthenticated, err.Error())
 	}
 
-	prefix, err := CreatePath(ctx, keyInfo.ProjectID, -1, req.BucketName, []byte(""))
+	// check if attribution is set for given bucket
+	_, err = endpoint.partnerinfo.Get(ctx, keyInfo.ProjectID, req.GetBucketName())
 	if err != nil {
-		return false, err
-	}
+		if attribution.ErrBucketNotAttributed.Has(err) {
+			//handle bucket attribution
+			prefix, err := CreatePath(ctx, keyInfo.ProjectID, -1, req.BucketName, []byte(""))
+			if err != nil {
+				return nil, err
+			}
 
-	items, _, err := endpoint.metainfo.List(ctx, prefix, string(""), string(""), true, 1, 0)
-	if err != nil {
-		return false, err
+			items, _, err := endpoint.metainfo.List(ctx, prefix, "", "", true, 1, 0)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(items) > 0 {
+				return nil, Error.New("Bucket(%s) , PartnerID(%s) cannot be attributed", string(req.BucketName), string(req.PartnerId))
```

Comment:
Use `%q`, partnerID should be an uuid, so casting it to a string will end up badly.

Reply:
uuid supports the stringer interface

Reply:
Here `req.PartnerId` is still a `[]byte` and `string(xyz)` doesn't use `xyz.String()`.

Reply:
updated